packages/react-router/src/useBlocker.tsx

```diff
@@ -3,25 +3,62 @@ import { useRouter } from './useRouter'
 import type { BlockerFn } from '@tanstack/history'
 import type { ReactNode } from './route'
 
+type Resolver = {
```

Comment:
Rename this to something that's more scoped to what it is like `BlockerFnResolver`, `BlockerResolver`, or something to that effect.